Services/Utilities/classes/class.ilUtil.php

```diff
@@ -4857,7 +4870,12 @@ public static function sendInfo($a_info = "",$a_keep = false)
 	*/
 	public static function sendFailure($a_info = "",$a_keep = false)
 	{
-		global $tpl;
+		global $tpl, $DIC;
+
+		if ($DIC->http()->request()->getQueryParams()["async"] && !$a_keep) {
```

Comment:
Here you suddenly use a string "async". Please note the the general meaning of doing something asynchronously and getting a JSON response are two very different things. I currently would expect both being set for these requests: $_REQUEST["cmdMode"] == "asynch" and something like $_REQUEST["responseMode"] == "json".